include/nbl/builtin/hlsl/blit/common.hlsl

```diff
@@ -0,0 +1,147 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_BLIT_COMMON_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BLIT_COMMON_INCLUDED_
+
+#include <nbl/builtin/hlsl/cpp_compat/cpp_compat.hlsl>
+#include <nbl/builtin/hlsl/blit/formats_encode.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace blit
+{
+
+template <uint32_t AlphaBinCount>
+struct AlphaStatistics
+{
+	uint32_t passedPixelCount;
+	uint32_t histogram [AlphaBinCount];
+};
```

Comment:
you simply cannot define this, it needs to be provided by the user because the values need to be `groupshared` and either global or `static` which in turn makes them consume shared memory and reduce occupancy for no reason if you don't use them.

This is why Pentakon's PR #519  requires them to be "provided from the outside" and there's a set "interface/concept" these shared acessors need to conform to, like providing a `uint32_t` set/get plus workgroup execution barriers.

Reply:
P.S. separate `REF` accessor per member, so in this case, one for the histogram and one for the passed pixel count

Reply:
Updated the comment, I forgot this was never in shared memory.